src/main/steps/c100-rebuild/child-details/other-children/names/postContoller.ts

```diff
@@ -0,0 +1,134 @@
+      const childrenDetailsErrors = req.session.errors.filter(
+        error => !['otherChildFirstName', 'otherChildLastName'].includes(error.propertyName)
+      );
+
+      if (req.session.userCase?.cd_otherChildren?.length) {
+        //if there are already children added
+        if (childrenDetailsErrors.length) {
+          //if there are errors present for added children, save the changes made on the input fields
+          req.session.userCase.cd_otherChildren = this.transformData(rest, req.session.userCase.cd_otherChildren);
+        } else {
+          //if there are no errors present for added children, then save the details
+          req.session.userCase.cd_otherChildren = this.transformData(rest, req.session.userCase.cd_otherChildren);
+          if (otherChildFirstName && otherChildLastName) {
+            // if first name & last name is fed for new child, add the new entry to the session
+            req.session.userCase.cd_otherChildren.push({
+              ...getDataShape(),
+              firstName: otherChildFirstName as string,
+              lastName: otherChildLastName as string,
+            });
+            Object.assign(req.session.userCase, { otherChildFirstName: '', otherChildLastName: '' });
+          } else if (!otherChildFirstName && !otherChildLastName) {
+            // if first name & last name is empty, remove errors from the session
+            req.session.errors = [];
+          }
+        }
+      } else {
+        //if there are no children added
+        if (otherChildFirstName && otherChildLastName) {
+          req.session.userCase.cd_otherChildren = [
+            {
+              ...getDataShape(),
+              firstName: otherChildFirstName as string,
+              lastName: otherChildLastName as string,
+            },
+          ];
+          Object.assign(req.session.userCase, { otherChildFirstName: '', otherChildLastName: '' });
+        }
+      }
+
+      return super.redirect(req, res);
+    } else if (saveAndComeLater) {
+      const dataToSave = { cd_otherChildren: [...this.transformData(rest, req.session.userCase.cd_otherChildren)] };
+
+      if (otherChildFirstName && otherChildLastName) {
+        dataToSave.cd_otherChildren.push({
+          ...getDataShape(),
+          firstName: otherChildFirstName,
+          lastName: otherChildLastName,
+        });
+        Object.assign(req.session.userCase, { otherChildFirstName: '', otherChildLastName: '' });
+      }
+
+      super.saveAndComeLater(req, res, dataToSave);
+    }
+  }
+
+  private transformData(formData, orginialData: ChildrenDetails[] = []): ChildrenDetails[] {
+    return Object.entries(formData).reduce(
+      (transformedData: ChildrenDetails[], [fieldName, value]) => {
+        const [fieldId, fieldIndex] = fieldName.split('-');
+        const index = Number(fieldIndex) - 1;
+
+        if (!transformedData[index]) {
+          transformedData[index] = getDataShape();
```

Comment:
You cannot simply use the getDataShape of child details here. Did you check the userCase for other children with what all properties getting created? Try to extend the base data shape for other children